spikeinterface/extractors/nwbextractors.py

```diff
@@ -176,18 +176,17 @@ def __init__(
             times_kwargs = dict(sampling_frequency=sampling_frequency, t_start=t_start)
 
         num_frames = int(self._es.data.shape[0])
-        num_channels = len(self._es.electrodes.data)
 
         # Extractors channel groups must be integers, but Nwb electrodes group_name can be strings
         if 'group_name' in self._nwbfile.electrodes.colnames:
             unique_grp_names = list(np.unique(self._nwbfile.electrodes['group_name'][:]))
 
         # Fill channel properties dictionary from electrodes table
         if "channel_name" in self._nwbfile.electrodes.colnames:
-            channel_ids = self._es.electrodes["channel_name"][:].astype("str")
+            channel_ids = [self._es.electrodes["channel_name"][i] for i in self._es.electrodes.data]
```

Comment:
What is "data" exactly here ?

Reply:
that's the only way I found to extract what seems to be the ids of the electrodes table associated with this electrical series

Reply:
I think that is a good solution. @CodyCBakerPhD could you take a look? Is the `es.electrodes`  a link to the `electrodes` table region used by the `ElectricalSeries`?

Reply:
The `data` field of a DynamicTableRegion is supposed to be the subregion of the target table that is subindexed, i.e.,

```python
nwbfile.acquisition[es_key].electrodes.data[:]
```

Reply:
I think you should keep the string. I remember that some round-trip tests failed in neuroconv because of this but I can't remember exactly which. At least try to run neuroconv tests with this branch.